### PR TITLE
[WIP] Rewrite the lifecycle doc and FAQs

### LINKED ISSUE #14596: Rewrite the lifecycle doc to include the API info
Include the high-level overview of the editor's API in the lifecycle page.

### PR DESCRIPTION
**Pull request by Witoso**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: Adding more information about API in the docs. Closes #14596.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: docs/installation/getting-started/editor-lifecycle.md [hunk 1]
```diff
@@ -3,167 +3,157 @@ category: getting-started
...
+	],
+});
 ```
 
-### Example – decoupled editor
+Plugins created in this way can do things after initialization (afterInit) as well as set up new editor UI components.
+TODO links to plugin interface and framework.
 
-Add the elements where CKEditor should initialize the toolbar and the editable to your page:
+### Using the API
```
Comment (apadol-cksource) [inline docs/installation/getting-started/editor-lifecycle.md:105]: I see this as a separate category - API and Events. We could leave the "editors lifecycle" as is then. We could split examples into sub-articles for better findability. Explain what command is, what writer is, and how to use events.

File: docs/examples/framework/how-tos.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
+---
+category: examples-framework
```
Comment (apadol-cksource) [inline docs/examples/framework/how-tos.md:2]: We need a new category as those examples aren't related just to the framework. TBD.

Comment (godai78) [reply docs/examples/framework/how-tos.md:2]: That would fit in FAQ, but it's kinda obscure section.

**Comment by godai78** [comment]
After merging in the latest `master`, there is still a link missing:

```
Error: Invalid internal links in /ckeditor5/38.1.1/updating/ckeditor4/ckeditor4-troubleshooting.html:
  * '../../support/faq.html#how-to-insert-some-content-into-the-editor' - this CKEditor 5 FAQ question
  * '../../support/faq.html#how-to-list-all-instances-of-the-editor' - this CKEditor 5 FAQ question
  ```

**Comment by godai78** [comment]
I suppose a rebase to `stable` would now be in order, too.

**Comment by Witoso** [comment]
We will do this at the very end. I will either create a new branch that originates in stable and checkout the files in the branch or some other way.

**Comment by Witoso** [comment]
Moved to #14772.

**Pull request closed without merging**